### PR TITLE
chore(travis): add node 0.12, Firefox 38.0.1, remove gmp

### LINKED ISSUE #2365: Add support for Node 0.12
See https://github.com/mozilla/fxa-auth-server/pull/920

It'd be super neat if we updated **fxa-content-server** to work with Node 0.12.x and io.js and update our travis.yml to run tests for those envs.

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #2365

### CODE REVIEW

**Comment by vladikoff** [comment]
WIP, do not merge

File: .travis.yml [hunk 1]
```diff
@@ -53,3 +53,9 @@ script:
   - grunt lint
   - travis_retry npm run test-travis
   - travis_retry npm run test-server
+
+matrix:
+  allow_failures:
+    - node_js: "0.12"
+    - node_js: "iojs"
+  fast_finish: true
```
Comment (shane-tomlinson) [inline .travis.yml]: My understanding is that fast_finish aborts if there are *any* failures, yet [travis_retry](https://github.com/mozilla/fxa-content-server/pull/2371/files#diff-354f30a63fb0907d4ad57269548329e3R54) retries 3 times. Am I confused here?

File: .travis.yml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 language: node_js
 node_js:
   - "0.10"
+  - "0.12"
+  - "iojs"
```
Comment (pdehaan) [inline .travis.yml]: Also, I believe io.js just released v2.0.0 yesterday. We may want to test against io.js v1 and v2 separately.
See https://github.com/mozilla/fxa-auth-db-server/pull/139/files

File: .travis.yml (hunk 1 above)
Comment (jrgm) [inline .travis.yml]: Demo: https://travis-ci.org/jrgm/allowed_failures

**Comment by vladikoff** [comment]
Seems like we have a few issues with the server tests, will need to fix it up

**Comment by jrgm** [comment]
> Seems like we have a few issues with the server tests, will need to fix it up

I think there is only one server test failure, which is for `POST /auth/v0/expect-gone`. That problem goes away if you upgrade `connect-restreamer@latest`; the issue is that `body-parser:json` coerces a zero-length POST to `{}`, but leaves the `Content-Length` at `0`. That then trips up `connect-restreamer@1.0.1`, but a fix in `1.0.2` corrects the `Content-Length` back to `2`. 

So, for the short term, I'd upgrade, and also `http-parser` FWIW. Longer-term, it may make sense to file a bug to remove this proxy, since AFAIK, it's only there for IE8, and we said goodbye to IE8.

**Comment by vladikoff** [comment]
the bug to remove the proxy is here https://github.com/mozilla/fxa-content-server/issues/2058

File: .travis.yml (hunk 2 above)
Comment (pdehaan) [inline .travis.yml]: Semi-related, but how will npm work when this gets merged?
The latest Node 0.12 includes npm@2.7.4 ([changelog](https://raw.githubusercontent.com/joyent/node/v0.12.2/ChangeLog))

Does this mean that we'll be downgrading Node 0.12's and iojs's npm _down_ to npm@2.4 (per line 22)?

Comment (vladikoff) [inline .travis.yml]: > Does this mean that we'll be downgrading Node 0.12's and iojs's npm down to npm@2.4

Yeap, should still work I think

**Comment by vladikoff** [comment]
updated!

**Comment by vladikoff** [comment]
```
CancelError: Timeout reached on main - auth & oauth server proxy - #auth server proxied POST request
Error: Timeout reached on main - auth & oauth server proxy - #auth server proxied POST request
  at Error  <anonymous>
  at new ErrorCtor  <node_modules/intern-geezer/node_modules/dojo/errors/create.js:13:21>
  at null._onTimeout  <node_modules/intern-geezer/lib/Test.js:177:38>
  at Timer.listOnTimeout [as ontimeout]  <timers.js:112:15>
1/3 tests failed

```

**Comment by vladikoff** [comment]
We are also getting a coveralls result for each passed test, should fix this up ..

**Comment by zaach** [comment]
Do we still need this in light of recent news?

**Comment by vladikoff** [comment]
@jrgm still getting errors on 0.10 with this update, pretty strange
```
fxa-content-server.server.proxy.INFO: proxying request: (POST) /auth/v0/expect-gone => http://127.0.0.1:9000/v0/expect-gone
FAIL: #auth server proxied POST request (1005ms)
CancelError: Timeout reached on main - auth & oauth server proxy - #auth server proxied POST request
Error: Timeout reached on main - auth & oauth server proxy - #auth server proxied POST request
  at Error  <anonymous>
  at new ErrorCtor  <node_modules/intern-geezer/node_modules/dojo/errors/create.js:13:21>
  at null._onTimeout  <node_modules/intern-geezer/lib/Test.js:177:38>
  at Timer.listOnTimeout [as ontimeout]  <timers.js:112:15>
1/3 tests failed
```

> Do we still need this in light of recent news?

I think we can add 0.12 and not worry about iojs for now. Thoughts?

**Comment by shane-tomlinson** [comment]
> I think we can add 0.12 and not worry about iojs for now. Thoughts?

Sounds reasonable to me.

**Comment by vladikoff** [comment]
This PR is not going well, closing until we remove the proxy

**Pull request closed without merging**